src/m365/spo/commands/tenant/tenant-commandset-set.ts

```diff
@@ -0,0 +1,223 @@
+        if (!args.options.newTitle &&
+          !args.options.listType &&
+          !args.options.clientSideComponentId &&
+          !args.options.clientSideComponentProperties &&
+          !args.options.webTemplate &&
+          !args.options.location) {
+          return 'Specify at least one property to update';
+        }
+
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (args.options.listType && SpoTenantCommandSetSetCommand.listTypes.indexOf(args.options.listType) < 0) {
+          return `${args.options.listType} is not a valid list type. Allowed values are ${SpoTenantCommandSetSetCommand.listTypes.join(', ')}`;
+        }
+
+        if (args.options.location && SpoTenantCommandSetSetCommand.locations.indexOf(args.options.location) < 0) {
+          return `${args.options.location} is not a valid location. Allowed values are ${SpoTenantCommandSetSetCommand.locations.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      await this.updateTenantWideExtension(appCatalogUrl, args.options, logger);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  public async getTenantCommandSet(logger: Logger, options: Options, requestUrl: string): Promise<number> {
+    if (this.verbose) {
+      logger.logToStderr(`Getting the tenant command set ${options.id}`);
+    }
+
+    const filter: string = `startswith(TenantWideExtensionLocation,'ClientSideExtension.ListViewCommandSet') and Id eq '${options.id}'`;
+
+    const reqOptions: CliRequestOptions = {
+      url: `${requestUrl}/items?$filter=${filter}`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    const listItemInstances: ListItemInstanceCollection = await request.get<ListItemInstanceCollection>(reqOptions);
+
+    if (listItemInstances.value.length === 0) {
+      throw 'The specified command set was not found';
+    }
+
+    return listItemInstances.value[0].Id;
+  }
+
+  private async updateTenantWideExtension(appCatalogUrl: string, options: Options, logger: Logger): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr('Updating tenant wide extension to the TenantWideExtensions list');
+    }
+
+    const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+    const requestUrl = `${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')`;
+    const id = await this.getTenantCommandSet(logger, options, requestUrl);
```

Comment:
We don't need to get the id as we already have it. What we want to do here is validate if the commandset exists. So we actually don't even need the function to return a value.

But if I were to choose I'd say let's refactor it into a general getListItemById function. (We can even move it to the spo util, later!). Let it return null if the item is not found. 

As soon as we have the listitem we can now check: whether it is null or if it's not a commandSet. Now the function and it's returnvalue is useful.

Reply:
the api returns an error if the listitem is not found, so we don't have to return null here or do errorhandling around this